test/e2e/playwright/swap/pageObjects/swap-page.ts

```diff
@@ -53,37 +58,73 @@ export class SwapPage {
     this.swapTokenButton = this.page.locator('button', { hasText: 'Swap' });
     this.closeButton = this.page.getByText('Close');
     this.backButton = this.page.locator('[title="Cancel"]');
+    this.viewInActivityBtn = this.page.getByTestId(
+      'page-container-footer-next',
+    );
   }
 
-  async fetchQuote(options: { from?: string; to: string; qty: string }) {
-    // Enter Swap Quantity
-    await this.tokenQty.fill(options.qty);
-
+  async enterQuote(options: { from?: string; to: string; qty: string }) {
     // Enter source token
-    if (options.from) {
+    const native = await this.page.$(`text=/${options.from}/`);
+    if (!native && options.from) {
       this.swapFromDropDown.click();
-      await this.tokenSearch.fill(options.from);
       await this.selectTokenFromList(options.from);
     }
 
-    // Enter destionation token
+    const balanceString = await this.page
+      .locator('[class*="balance"]')
+      .first()
+      .textContent();
+    if (balanceString) {
+      if (parseFloat(balanceString.split(' ')[1]) <= parseFloat(options.qty)) {
+        await this.gotBack();
+        // not enough balance so cancel out
+        return false;
+      }
+    }
+
+    // Enter Swap Quantity
+    await this.tokenQty.fill(options.qty);
+    this.swapQty = options.qty;
+
+    // Enter destination token
     await this.swapToDropDown.click();
-    await this.tokenSearch.fill(options.to);
     await this.selectTokenFromList(options.to);
+    return true;
+  }
+
+  async waitForQuote() {
+    let quoteFound = false;
+    do {
+      // Clear Swap Anyway button if present
+      const swapAnywayButton = await this.page.$('text=/Swap anyway/');
+      if (swapAnywayButton) {
+        await swapAnywayButton.click();
+      }
+
+      // No quotes available
+      const noQuotes = await this.page.$('text=/No quotes available/');
+      if (noQuotes) {
+        await this.gotBack();
```

Comment:
Typo in the method call `gotBack()`, origin class method should be renamed to`goBack()`.

Reply:
done